src/blocks/form/edit.js

```diff
@@ -279,17 +273,22 @@ class FormEdit extends Component {
 	setTemplate( layout ) {
 		const { setAttributes } = this.props;
 		let submitButtonText;
+		let clientID = this.props.clientId;
 		map( TEMPLATE_OPTIONS, ( elem ) => {
 			if ( isEqual( elem.template, layout ) ) {
+				// // Update the child block's attributes
 				submitButtonText = elem.submitButtonText;
 				if ( Array.isArray( submitButtonText ) ) {
 					submitButtonText = submitButtonText.join( '' );
 				}
+				setTimeout( function() {
+					let childBlocks = wp.data.select( 'core/editor' ).getBlocksByClientId( clientID )[0].innerBlocks;
+					wp.data.dispatch('core/editor').updateBlockAttributes( childBlocks[ childBlocks.length - 1 ].clientId, { submitButtonText: submitButtonText } );
+				}, 100 );
 			}
 		} );
 
 		this.setState( { template: layout } );
-		setAttributes( { submitButtonText } );
 	}
```

Comment:
There is some strange behavior within `setTemplate`. We shouldn't need to track the template in state since any inner block changes are tracked by the InnerBlocks component. I think we can rip out this state property and maintain undo/redo states so long as we are handling when innerBlocks is empty.